File: C++/404. Sum of Left Leaves.cpp
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(root == nullptr)
        return 0;        
        int sum = 0;
        if(root->left !=nullptr && root->left->left == nullptr&& root->left->right == nullptr)
        {
            sum += root->left->val;        
        }        
        sum += sumOfLeftLeaves(root->left);
        sum += sumOfLeftLeaves(root->right);        
        return sum;
    }
};

File: JAVA/404. Sum of Left Leaves.java
class Solution {
    int value = 0;
    public int sumOfLeftLeaves(TreeNode root) {
        process(root);
        return value;
    }
    public void process(TreeNode root)
    {
        if (root!=null)
        {
            if (root.left != null && root.left.left == null && root.left.right == null)
                value += root.left.val;
            process(root.left);
            process(root.right);
        }
    }
}
